village/src/App.js

```diff
@@ -11,14 +17,30 @@ class App extends Component {
       smurfs: [],
     };
   }
-  // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
-  // Notice what your map function is looping over and returning inside of Smurfs.
-  // You'll need to make sure you have the right properties on state and pass them down to props.
+  
+componentDidMount() {
+  axios
+    .get('http://localhost:3333/smurfs')
+    .then(res => this.setState({ smurfs: res.data }))
+    .catch(err => console.log(err))
+}
+
+addSmurf = (e, smurf) => {
+  e.preventDefault();
+  const {name, age, height} = smurf
+
+  axios
+    .post('http://localhost:3333/smurfs', {name, age, height})
+    .then(res => this.setState({ smurf: res.data }))
```

Comment:
Found out why your list of characteres doesn't rerender when you add a smurf. 😄 
```suggestion
    .then(res => this.setState({ smurfs: res.data }))
```